ignite/src/jepsen/ignite.clj

```diff
@@ -0,0 +1,161 @@
+    "Creates a config file."
+    (let [tmp-file (File/createTempFile "jepsen-ignite-config" ".xml")
+          tmp-file-path (.getCanonicalPath tmp-file)]
+        (spit tmp-file-path (clojure.string/replace (slurp "resources/apache-ignite.xml.tmpl")
+            #"##addresses##|##instancename##|##clientmode##"
+            { "##addresses##" (clojure.string/join "\n" (map #(str "\t<value>" % ":47500..47509</value>") addresses))
+            "##instancename##" (.getName tmp-file)
+            "##clientmode##" (str client-mode) }))
+        tmp-file))
+
+(defn configure-server [addresses node]
+    "Creates a server config file and uploads it to the given node."
+    (let [src-file (configure addresses false)]
+        (c/upload src-file (str server-dir "server-ignite-" node ".xml"))
+        (.delete src-file)))
+
+(defn configure-client [addresses]
+    "Creates a client config file."
+    (configure addresses true))
+
+(defn db
+    "Apache Ignite cluster life-cycle."
+    [version]
+    (reify db/DB
+        (setup! [_ test node]
+            (info node "Installing Apache Ignite" version)
+            (debian/install-jdk8!)
+            (cu/ensure-user! user)
+            (let [url (str "https://archive.apache.org/dist/ignite/" version "/apache-ignite-" version "-bin.zip")]
+                (cu/install-archive! url server-dir))
+            (configure-server (:nodes test) node)
+            (c/exec :chown :-R (str user ":" user) server-dir)
+            (start! node test)
+            (await-node-started node test)
+            (jepsen/synchronize test))
+
+        (teardown! [_ test node]
+            (nuke! node test))
+
+        db/LogFiles
+            (log-files [_ test node]
+            [logfile])))
+
+(defn r   [_ _] {:type :invoke, :f :read, :value nil})
+(defn w   [_ _] {:type :invoke, :f :write, :value (rand-int 1000)})
+
+(defrecord Client [conn cache config]
+    client/Client
+    (open! [this test node]
+        (let [config (configure-client (:nodes test))
+              conn (Ignition/start (.getCanonicalPath config))
+              cache (.getOrCreateCache conn "JepsenCache")]
+            (assoc this :conn conn :cache cache :config config)))
+
+    (setup! [this test])
+
+    (invoke! [_ test op]
+        (try
+            (case (:f op)
+                :read (let [value (.get cache "k")]
+                    (assoc op :type :ok :value value))
+                :write (do (.put cache "k" (:value op))
+                    (assoc op :type :ok)))))
+
+    (teardown! [this test]
+        (.delete config))
+
+    (close! [_ test]
+        (.stop conn true)))
+
+(defn ignite-test
+    "Given an options map from the command-line runner (e.g. :nodes, :ssh,
+    :concurrency, ...), constructs a test map."
+    [opts]
+    (info :opts opts)
+    (merge tests/noop-test
+        opts
+        {:name "ignite"
+        :os debian/os
+        :db (db "2.7.0")
+        :client (Client. nil nil nil)
+        :concurrency 3
```

Comment:
Why hardcode the concurrency to 3 here? This breaks the --concurrency option from the CLI, and since you've got five nodes, you won't be making queries to 2 of them.